src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperAttribute.cs

```diff
@@ -77,8 +79,130 @@ public bool Equals(TagHelperAttribute other)
             return
                 other != null &&
                 string.Equals(Name, other.Name, StringComparison.OrdinalIgnoreCase) &&
-                Minimized == other.Minimized &&
-                (Minimized || Equals(Value, other.Value));
+                Structure == other.Structure &&
+                (Structure == HtmlAttributeStructure.Minimized || Equals(Value, other.Value));
+        }
+
+        /// <inheritdoc />
+        public void WriteTo(TextWriter writer, HtmlEncoder encoder) => WriteTo(Name, Value, Structure, writer, encoder);
+
+        /// <inheritdoc />
+        public void CopyTo(IHtmlContentBuilder destination) => CopyTo(Name, Value, Structure, destination);
+
+        /// <inheritdoc />
+        public void MoveTo(IHtmlContentBuilder destination) => CopyTo(destination);
+
+        /// <summary>
+        /// TODO
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="value"></param>
+        /// <param name="structure"></param>
+        /// <param name="destination"></param>
+        public static void CopyTo(
+            string name,
+            object value,
+            HtmlAttributeStructure structure,
+            IHtmlContentBuilder destination)
```

Comment:
Do not add these `public static` methods. This is an immutable `class` that knows how to write _itself_ out.